src/scripts/main.js

```diff
@@ -1,5 +1,31 @@
 'use strict';
 
-document.addEventListener('click', e => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+wall.addEventListener('click', e => {
+  const workSpaceY = wall.clientHeight - spider.clientHeight;
+  const workSpaceX = wall.clientWidth - spider.clientWidth;
+
+  let normalizedY = e.offsetY;
+  let normalizedX = e.offsetX;
+
+  if (e.offsetY < 0) {
+    normalizedY = spider.clientHeight;
+  }
+
+  if (e.offsetY > workSpaceY) {
+    normalizedY = workSpaceY;
+  }
+
+  if (e.offsetX < 0) {
+    normalizedX = spider.clientWidth;
```

Comment:
`spider.clientWidth` is not the best value here: I can't move the spider close to the border clicking on left and top border:
![image](https://user-images.githubusercontent.com/86923089/219434087-9965c1d8-1341-43c1-b221-c86d44a85e32.png)
![image](https://user-images.githubusercontent.com/86923089/219434149-ae462613-8e1c-4dfe-8b6b-eb58f0919241.png)